### ISSUE TITLE
Image pull secrets not set on node agent DaemonSet

### ISSUE THREAD

**Issue by Bondza** [DataDog/datadog-operator#802]
**Describe what happened:**
We are trying to use the Datadog operator to manage the node agent in our clusters. It works well while using the default image settings, but we use our own registry for all images and therefore require `pullSecrets` to be set to be able to pull images. To do this we set `spec.override.nodeAgent.image.pullSecrets` like:
```yaml
apiVersion: datadoghq.com/v2alpha1
kind: DatadogAgent
metadata:
  name: datadog
  namespace: default
spec:
  ...
  override:
    nodeAgent:
      image:
        name: image-name-from-our-registry:version
        pullSecrets:
        - name: secret-name
```

But the agent pods fail because they cannot pull the image from our registry:
```sh
> kubectl get pods -l app.kubernetes.io/instance=datadog-agent
NAME                  READY   STATUS                  RESTARTS   AGE
datadog-agent-c9727   0/1     Init:ImagePullBackOff   0          52m
...
datadog-agent-khdkp   0/1     Init:ErrImagePull       0          52m
```

And `imagePullSecrets` are missing from the DaemonSet.
```sh
> kubectl get daemonset datadog-agent -o jsonpath='{.metadata.name} {.spec.template.spec.imagePullSecrets}'
datadog-agent 
```

**Describe what you expected:**
I expected `imagePullSecrets` to be set on the DaemonSet so that the image can be pulled from a private registry.

**Steps to reproduce the issue:**
Apply a DatadogAgent manifest with `spec.override.nodeAgent.image.pullSecrets` set.

Check the DaemonSet for `imagePullSecrets`
```sh
> kubectl get daemonset datadog-agent -o jsonpath='{.metadata.name} {.spec.template.spec.imagePullSecrets}'
datadog-agent 
```

**Comment by ivanov-danil** [comment]
Same story. Also with deployment cluster-agent. 

```
nodeAgent:
  image:
    pullSecrets:
    - name: registrysecret

```
```
clusterAgent:
  image:
    pullSecrets:
    - name: registrysecret
```

**Comment by celenechang** [comment]
Hi @Bondza and @ivanov-danil thanks for reporting. I just merged PR #824 824 to fix this issue, it will be available in the next release (1.1.0).
